src/Rule/Trait/WhenTrait.php

```diff
@@ -23,7 +42,12 @@ public function when(Closure|null $value): static
     }
 
     /**
-     * @psalm-return WhenType
+     * A getter for `$when` property.
+     *
+     * @psalm-return WhenType Current value:
```

Comment:
For correct work documentation generators we can make base phpdoc annotations (`@return`, `@param`, ...) with description and psalm annotations without description. For example:

```
     *
     * @return Closure|null Current value:
     * 
     * - `null` - always apply the validation.
     * - `callable` - apply the validation depending on a return value: `true` - apply, `false` - do not apply.
     *
     * @psalm-return WhenType
     *
````

`@return` for humans and generators.

`@psalm-return` for psalm only.